plugins/performance-lab/includes/admin/rest-api.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * REST API integration for the plugin.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Namespace for performance-lab REST API.
+ *
+ * @since n.e.x.t
+ * @var string
+ */
+const PERFLAB_REST_API_NAMESPACE = 'performance-lab/v1';
+
+/**
+ * Route for activating plugin/feature.
+ *
+ * Note the `:activate` art of the endpoint follows Google's guidance in AIP-136 for the use of the POST method in a way
+ * that does not strictly follow the standard usage.
+ *
+ * @since n.e.x.t
+ * @link https://google.aip.dev/136
+ * @var string
+ */
+const PERFLAB_FEATURES_ACTIVATE_ROUTE = '/features/(?P<slug>[a-z0-9_-]+):activate';
+
+/**
+ * Route for fetching plugin/feature information.
+ *
+ * @since n.e.x.t
+ * @var string
+ */
+const PERFLAB_FEATURES_INFORMATION_ROUTE = '/features/(?P<slug>[a-z0-9_-]+)';
+
+/**
+ * Registers endpoint for performance-lab REST API.
+ *
+ * @since n.e.x.t
+ * @access private
+ */
+function perflab_register_endpoint(): void {
+	register_rest_route(
+		PERFLAB_REST_API_NAMESPACE,
+		PERFLAB_FEATURES_ACTIVATE_ROUTE,
+		array(
+			'methods'             => 'POST',
+			'args'                => array(
+				'slug' => array(
+					'type'              => 'string',
+					'description'       => __( 'Plugin slug of plugin/feature that needs to be activated.', 'performance-lab' ),
+					'required'          => true,
+					'validate_callback' => 'perflab_validate_slug_endpoint_arg',
```

Comment:
Since this validates against the list of PL features/plugins, should we include an `enum` entry here with the possible values?

If that feels like too much, I think we should at least mention it in the description, e.g. "Must be one of the Performance Lab feature slugs."

Reply:
I initially proposed using an `enum` here, but it won't work unless we then also always `require` the additional admin-only PHP files used in PerfLab. So what is implemented with this `validate_callback` is a sort of just-in-time enum. This is already explained in the phpdoc for `perflab_validate_slug_endpoint_arg()`.

Reply:
SG. In that case let's just mention it in the description.

Reply:
Like so? https://github.com/WordPress/performance/pull/1646/files#r1844278835